### PR TITLE
fixed multiple highlights and improved leaderboard refresh

### PR DESCRIPTION
**Pull request by lkaijie**
# Description
If u furiously scroll the leaderboards it would result in multiple users being highlighted yellow, because apparently the views in listview are recycled or something like that, which causes this. Fixed this, also added a refresh button instead of scroll up to refresh.
# Screenshot
Here is the refresh button,

[bugfix.webm](https://user-images.githubusercontent.com/94023052/225781066-49801072-560c-4005-94ea-b7e83997627e.webm)

# Additional Context
Use recyclerview
I should have used recyclerview
maybe ill switch it to a recyclerview

### CODE REVIEW

File: QRHunter/app/src/main/java/com/example/qrhunter/ui/profile/ProfileViewModel.java [hunk 1]
```diff
@@ -56,7 +57,7 @@ public void removeScannedQRCode(String qrCodeId, String playerId) {
         // Update this.scannedQRCodes
         ArrayList<QRCode> currentScannedQRCodes = this.scannedQRCodes.getValue();
         currentScannedQRCodes.removeIf(qrCode -> {
-            if (qrCode.getId() == qrCodeId) {
+            if (Objects.equals(qrCode.getId(), qrCodeId)) {
```
Comment (MaxxonTan) [inline QRHunter/app/src/main/java/com/example/qrhunter/ui/profile/ProfileViewModel.java:60]: Was this causing an issue before?

Comment (MaxxonTan) [inline QRHunter/app/src/main/java/com/example/qrhunter/ui/profile/ProfileViewModel.java:60]: Was this causing an issue before?

**Review by MaxxonTan — APPROVED** [review]
Looks good, just add a screen recording and small question

File: QRHunter/app/src/main/java/com/example/qrhunter/ui/profile/ProfileViewModel.java (hunk 1 above)
Comment (lkaijie) [reply QRHunter/app/src/main/java/com/example/qrhunter/ui/profile/ProfileViewModel.java:60]: nope, i must have changed it when i was trying to figure out what the issue was, it should do the same thing tho.

**Comment by lkaijie** [comment]
> Can you include a gif recording? Use https://www.screentogif.com/ if you're not sure how to do gif recording

i don't have the bugged version anymore, or do u mean the recording with the new refresh button?

File: QRHunter/app/src/main/java/com/example/qrhunter/ui/profile/ProfileViewModel.java (hunk 1 above)
Comment (MaxxonTan) [reply QRHunter/app/src/main/java/com/example/qrhunter/ui/profile/ProfileViewModel.java:60]: Hmm this is checking whether they both point to the same object reference, i'm not sure if it'll still work

**Comment by MaxxonTan** [comment]
> > Can you include a gif recording? Use https://www.screentogif.com/ if you're not sure how to do gif recording
> 
> i don't have the bugged version anymore, or do u mean the recording with the new refresh button?

No yea the screenrecording works, you should always include one if applicable so anyone reviewing don't have to go to your branch and see what the changes are visually

File: QRHunter/app/src/main/java/com/example/qrhunter/ui/profile/ProfileViewModel.java (hunk 1 above)
Comment (lkaijie) [reply QRHunter/app/src/main/java/com/example/qrhunter/ui/profile/ProfileViewModel.java:60]: i think (qrCode.getId() == qrCodeId) checks if it points to same object reference and object.equals compares the string itself, but i can change it

Comment (MaxxonTan) [reply QRHunter/app/src/main/java/com/example/qrhunter/ui/profile/ProfileViewModel.java:60]: Oh you're right, I got it the other way around mb

**Comment by lkaijie** [comment]
> recording works, you should always include



> > > Can you include a gif recording? Use https://www.screentogif.com/ if you're not sure how to do gif recording
> > 
> > 
> > i don't have the bugged version anymore, or do u mean the recording with the new refresh button?
> 
> No yea the screenrecording works, you should always include one if applicable so anyone reviewing don't have to go to your branch and see what the changes are visually

ok will do next time 👍

**Review by vtrieu5 — APPROVED** [review]

**Pull request merged**